src/processor/model/mod.rs

```diff
@@ -0,0 +1,162 @@
+pub mod otel_transform;
+mod py_transform;
+
+use crate::processor::py::PyResourceSpans;
+use pyo3::prelude::*;
+use std::ffi::CString;
+use std::sync::{Arc, Mutex};
+use tower::BoxError;
+use tracing::error;
+
+#[derive(Debug, Clone)]
+pub struct AnyValue {
+    pub value: Arc<Mutex<Option<Value>>>,
+}
+
+#[derive(Debug, Clone)]
+#[allow(clippy::enum_variant_names)]
+pub enum Value {
+    StringValue(String),
+    BoolValue(bool),
+    IntValue(i64),
+    DoubleValue(f64),
+    ArrayValue(ArrayValue),
+    KvListValue(KeyValueList),
+    BytesValue(Vec<u8>),
+}
+
+#[derive(Debug, Clone)]
+pub struct ArrayValue {
+    pub values: Vec<AnyValue>,
+}
+
+#[derive(Debug, Clone)]
+pub struct KeyValueList {
+    pub values: Vec<opentelemetry_proto::tonic::common::v1::KeyValue>,
+}
+
+#[derive(Debug, Clone)]
+pub struct KeyValue {
+    pub key: Arc<Mutex<String>>,
+    pub value: Arc<Mutex<Option<AnyValue>>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Resource {
+    pub attributes: Arc<Mutex<Vec<Arc<Mutex<KeyValue>>>>>,
+    pub dropped_attributes_count: Arc<Mutex<u32>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ResourceSpans {
+    pub resource: Arc<Mutex<Option<Resource>>>,
+    pub scope_spans: Arc<Mutex<Vec<Arc<Mutex<ScopeSpans>>>>>,
+    pub _schema_url: Arc<Mutex<String>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ScopeSpans {
+    //pub scope: ::core::option::Option<super::super::common::v1::InstrumentationScope>,
+    pub spans: Arc<Mutex<Vec<Arc<Mutex<Span>>>>>,
+    pub schema_url: Arc<Mutex<String>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Span {
+    pub trace_id: Vec<u8>,
+    pub span_id: Vec<u8>,
+    pub trace_state: String,
+    pub parent_span_id: Vec<u8>,
+    pub flags: u32,
+    pub name: String,
+    pub kind: i32,
+    pub start_time_unix_nano: u64,
+    pub end_time_unix_nano: u64,
+    pub attributes: Arc<Mutex<Vec<Arc<Mutex<KeyValue>>>>>,
+    pub dropped_attributes_count: u32,
+    //pub events: ::prost::alloc::vec::Vec<span::Event>,
+    pub dropped_events_count: u32,
+    //pub links: ::prost::alloc::vec::Vec<span::Link>,
+    pub dropped_links_count: u32,
+    pub status: Arc<Mutex<Option<Status>>>,
+}
+#[derive(Debug, Clone)]
+pub struct Status {
+    pub message: String,
+    pub code: i32,
+}
+
+pub fn register_processor(code: String, script: String, module: String) -> Result<(), BoxError> {
```

Comment:
Does this store the module internally or is this mostly to test that it compiles?

Reply:
So it actually stores it internally and it's a small optimization to create the module every time. Below you'll see we just import the cached module at line 120 `let py_mod = PyModule::import(py, processor)?;`